docs/dev/run_the_etl.rst

```diff
@@ -11,24 +11,192 @@ just want to use already processed data.
+
+    *IO Managers are user-provided objects that store asset outputs
+    and load them as inputs to downstream assets.*
+
+Each asset has an `IO Manager  <https://docs.dagster.io/concepts/io-management/io-managers>`__
+that tells Dagster how to handle the objects returned by the software
+defined asset's underlying function. The IO Managers in PUDL read and
+write dataframes to and from sqlite, pickle and parquet files. For example,
+the :func:`pudl.io_managers.pudl_sqlite_io_manager` allows assets to read
+and write dataframes and execute SQL statements.
+
+**Resources:**
+`Resources <https://docs.dagster.io/concepts/resources>`__ are objects
+that can be shared across multiple software-defined assets.
+For example, multiple PUDL assets use the :func:`pudl.resources.datastore`
+resource to pull data from PUDL's raw data archives on Zenodo.
+
+Generally, inputs to assets should either be other assets or
+python objects in Resources.
+
+**Jobs**:
+`Jobs <https://docs.dagster.io/concepts/ops-jobs-graphs/jobs>`__
+are preconfigured collections of assets, resources and IO Managers.
+Jobs are the main unit of execution in Dagster. For example,
+the ``etl_fast`` job defined in :mod:`pudl.etl` executes the
+FERC, EIA and EPA CEMS pipelines for the most recent year.
+
+**Definitions**:
+`Definitions  <https://docs.dagster.io/concepts/code-locations>`__
+are collections of assets, resources, IO managers and jobs that can
+be loaded into dagit and executed. Definitions can have multiple
+preconfigured jobs. For example, the ``pudl.ferc_to_sqlite`` definition
+contains ``etl_fast`` and ``etl_full`` jobs.
+
+There are two main Definitions in the PUDL processing pipeline:
+
+1. :func:`pudl.ferc_to_sqlite.defs` :doc:`converts the FERC Form 1, 2, 6, 60 and
+   714 DBF/XBRL files <clone_ferc1>` into `SQLite <https://sqlite.org>`__
+   databases so that the data are easier to extract, and so all of the raw FERC
+   data is available in a modern format. You must run a job in this definition
+   before you can execute a job in :func:`pudl.etl.defs`.
+2. :func:`pudl.etl.defs` coordinates the "Extract, Transform, Load" process that
+   processes 20+ years worth of data from the FERC Form 1 database, dozens of EIA
+   spreadsheets, and the thousands of CSV files that make up the EPA CEMS hourly
+   emissions data into a clean, well normalized SQLite database (for the FERC and
+   EIA data), and an `Apache Parquet <https://parquet.apache.org/>`__ dataset that
+   is partitioned by state and year (for the EPA CEMS).
+
+Both definitions have two preconfigured jobs:
+  - ``etl_fast`` processes one year of data
+  - ``etl_full`` processes all years of data
+
+.. _run-dagit:
+
+Running the ETL with Dagit
+--------------------------
+Dagster needs a directory to store run logs and some assets. Create a new directory
+outside of the pudl respository directory called ``dagster_home/``. Then set the
+``DAGSTER_HOME`` environment variable to the path of the new directory:
```

Comment:
Good question. DAGSTER_HOME mostly stores logs and outputs from assets that use the default [fs_io_manager](https://docs.dagster.io/_apidocs/io-managers#dagster.fs_io_manager). The fs_io_manager is used by assets that we aren't persisting to parquet or sqlite files. By default, the fs_io_manager stores the asset outputs as pickle files in the $DASTGER_HOME/storage/ directory. We could configure fs_io_manager to dump the pickle files into the PUDL_OUTPUT directory. This might clutter up the PUDL_OUTPUT directory with pickle files of partially cleaned tables we haven't documented. 

We still need to figure out how to distribute these raw / interim tables. I imagine our use of the fs_io_manager will decline as we start to persist and document these types of tables.

Reply:
Or at least explain here how it differs from PUDL_OUTPUTS in the meantime.

Reply:
One nice thing about Dagster is we don't need to know where this information is being stored to access it. If you check out the the `inspect-assets.ipynb` notebook, it shows how to access an asset with nothing but its AssetKey (ID). This is more robust than thinking about where exactly it's been stored, as it will continue to work without any change as we move the assets from one kind of storage to another (e.g. migrate interim outputs into the DB over time).

Reply:
cool!